From 005a99fc5c8c926e30bf2129704dbe0e975678f2 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 22 Jun 2021 19:50:27 +0300
Subject: [PATCH] [SW-0] module_lkd: add err msg on wfe denial

If security is enabled, the WFE command is denied, but no log is printed
about it.
Add an error message for this case.

Change-Id: I6f7d7dc8b8c85e7127a34f914e313d53ea8fedcf
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 8e2524c..3e6a709 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -252,6 +252,7 @@
 void module_lkd_run_fast(void)
 {
 	const char *cmd_str = NULL;
+	const char *err_str = NULL;
 
 	switch (sts.cmd.cmd) {
 	case COMMS_NOOP:
@@ -282,10 +283,12 @@
 		break;
 	case COMMS_GOTO_WFE:
 		cmd_str = "wfe";
-		if (preboot_security_is_enabled())
+		if (preboot_security_is_enabled()) {
 			module_lkd_send_status(COMMS_STS_ERR, COMMS_SRAM, 0);
-		else
+			err_str = "WFE denied - security is enabled...";
+		} else {
 			module_lkd_send_status(COMMS_STS_ACK, COMMS_SRAM, 0);
+		}
 
 		break;
 	case COMMS_SKIP_BMC:
@@ -304,6 +307,11 @@
 	preboot_puts_raw("Got host ");
 	preboot_puts_raw(cmd_str);
 	preboot_puts(" command\n");
+
+	if (err_str) {
+		preboot_puts_raw(err_str);
+		preboot_puts("\n");
+	}
 }
 
 static int module_lkd_msg_validate(uintptr_t offset)
